consensus/tendermint/node_view.go

```diff
@@ -14,31 +14,42 @@ import (
 )
 
 type NodeView struct {
-	tmNode             *Node
-	validatorPublicKey crypto.PublicKey
-	txDecoder          txs.Decoder
-	runID              simpleuuid.UUID
+	tmNode    *Node
+	publicKey crypto.PublicKey
+	txDecoder txs.Decoder
+	runID     simpleuuid.UUID
 }
 
-func NewNodeView(tmNode *Node, txDecoder txs.Decoder, runID simpleuuid.UUID) (*NodeView, error) {
-	publicKey, err := crypto.PublicKeyFromTendermintPubKey(tmNode.PrivValidator().GetPubKey())
-	if err != nil {
-		return nil, err
+func NewNodeView(txDecoder txs.Decoder, runID simpleuuid.UUID) *NodeView {
+	return &NodeView{
+		txDecoder: txDecoder,
+		runID:     runID,
 	}
+}
+
+func (nv *NodeView) SetValidator(privVal types.PrivValidator) error {
+	pK, err := crypto.PublicKeyFromTendermintPubKey(privVal.GetPubKey())
+	nv.publicKey = pK
+	return err
+}
+
+func (nv *NodeView) SetNode(tmNode *Node) {
 	tmNode.BlockStore()
-	return &NodeView{
-		validatorPublicKey: publicKey,
-		tmNode:             tmNode,
-		txDecoder:          txDecoder,
-		runID:              runID,
-	}, nil
+	nv.tmNode = tmNode
+}
+
+func (nv *NodeView) GetNode() *Node {
+	return nv.tmNode
 }
 
 func (nv *NodeView) ValidatorPublicKey() crypto.PublicKey {
-	return nv.validatorPublicKey
+	return nv.publicKey
 }
 
 func (nv *NodeView) NodeInfo() *NodeInfo {
+	if nv.tmNode == nil {
```

Comment:
So I think the way we should handle this type of thing is being keeping `*NodeView` itself nil so:

```go
if nv == nil {
// return suitable no-op thing
}
```